internal/ackhandler/sent_packet_handler.go

```diff
@@ -348,15 +353,19 @@ func (h *sentPacketHandler) onPacketAcked(packetElement *PacketElement) {
 }
 
 func (h *sentPacketHandler) DequeuePacketForRetransmission() *Packet {
-	if len(h.retransmissionQueue) == 0 {
-		return nil
+	for len(h.retransmissionQueue) > 0 {
```

Comment:
By the time the packet ends up in the retransmission queue, we've already applied penalties due to the loss. Could that be a performance issue?

Reply:
Yes, and I don't really know what to do about this. Probably this will have to wait for the rewrite of the loss recovery.

Reply:
So the right way to solve this is to keep track of which packet is a retransmission of which one (which will be part of the rewrite). As soon as *any one* of the Initials (the original one or the retransmission) gets acked, we would remove *all* of them from the history.
I'm not yet entirely sure what to do with the retransmission queue. It seems to be a general case, that we need to cancel a retransmission, even if it's queued.